packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,1130 @@
+
+  // After a character is inserted when autocomplete is true and
+  // allowFreeform is false, remember the task that will clear
+  // the pending string of characters
+  private _lastReadOnlyAutoCompleteChangeTimeoutId: number;
+
+  // Promise used when resolving the comboBox options
+  private currentPromise: PromiseLike<IComboBoxOption[]>;
+
+  constructor(props?: IComboBoxProps) {
+    super(props);
+
+    this._warnMutuallyExclusive({
+      'defaultSelectedKey': 'selectedKey'
+    });
+
+    this._id = props.id || getId('ComboBox');
+
+    let selectedKey = props.defaultSelectedKey !== undefined ? props.defaultSelectedKey : props.selectedKey;
+    this._lastReadOnlyAutoCompleteChangeTimeoutId = -1;
+
+    let index: number = this._getSelectedIndex(props.options, selectedKey);
+
+    this.state = {
+      isOpen: false,
+      selectedIndex: index,
+      focused: false,
+      suggestedDisplayValue: '',
+      currentOptions: this.props.options,
+      currentPendingValueValidIndex: -1,
+      currentPendingValue: ''
+    };
+  }
+
+  public componentDidMount() {
+    // hook up resolving the options if needed on focus
+    this._events.on(this._comboBoxWrapper, 'focus', this._onResolveOptions, true);
+  }
+
+  public componentWillReceiveProps(newProps: IComboBoxProps) {
+    // In controlled component usage where selectedKey is provided, update the selectedIndex
+    // and currentOptions state if the key or options change
+    if (newProps.selectedKey !== undefined &&
+      (newProps.selectedKey !== this.props.selectedKey || newProps.options !== this.props.options)) {
+      let index: number = this._getSelectedIndex(newProps.options, newProps.selectedKey);
+      this.setState({
+        selectedIndex: index,
+        currentOptions: newProps.options
+      });
+    }
+  }
+
+  public componentDidUpdate(prevProps: IComboBoxProps, prevState: IComboBoxState) {
+    let {
+      allowFreeform
+    } = this.props;
+    let {
+      isOpen,
+      focused,
+      selectedIndex
+    } = this.state;
+
+    // If we are open, make sure the currently
+    // selected option is scrolled into view
+    if (isOpen) {
+      this._scrollIntoView();
+    }
+
+    // If we are open or we are focused but are not the activeElement,
+    // set focus on the input
+    if (isOpen || (focused && document.activeElement !== this._comboBox.inputElement)) {
+      this.focus();
+    }
+
+    // If we just opened/closed the menu OR
+    // updated the selectedIndex with the menu closed OR
+    // we are focused and are not allowing freeform
+    // we need to fix up focus and set selection
+    if (prevState.isOpen !== isOpen ||
+      (!isOpen && prevState.selectedIndex !== selectedIndex) ||
+      (!allowFreeform && focused)) {
+      this._select();
+    }
+  }
+
+  public componentWillUnmount() {
+    // remove the eventHanlder that was added in componentDidMount
+    this._events.off(this._comboBoxWrapper);
```

Comment:
actually you probably need to call the base explicitly now that you are overriding otherwise the this._async method calls you are enqueuing won't be cancelled in the case of unmount